### PR TITLE
Authentication concerns

### PR DESCRIPTION
**Pull request by vogdb**
Authentication model is very simple and basically allows everyone within organization to read and create jobs.

We probably would like to allow reading to devs, but create jobs mostly via hubot. 
To consider and implement.

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review.
I will add Docu after you the implemented Authorization is accepted.
@njam FYI.

File: lib/config.js [hunk 1]
```diff
@@ -85,6 +85,25 @@ var validScheme = {
         baseUrl: {
           type: 'uri',
           required: true
+        },
+        authorization: {
+          type: 'object',
+          properties: {
+            role: {
```
Comment (tomaszdurka) [inline lib/config.js]: Maybe we can skip this one?

**Comment by tomaszdurka** [comment]
- make restrictions a `authentication.installHandlers` parameter
- remove static hardcode dependecy
- convert `UserContainer._userList` to `Array`
- move `config.authentication.authorization.role.*` to `config.authentication.authorization.*`

**Comment by vogdb** [comment]
@tomaszdurka please review.

**Comment by vogdb** [comment]
@tomaszdurka please review.

**Comment by vogdb** [comment]
@tomaszdurka please review.

**Comment by vogdb** [comment]
Hello @tomaszdurka !

**Comment by tomaszdurka** [comment]
Sorries for delay - everytime I came back to this isue I still had some thoughts about if this is right place to put restrictions. I thought it should be maybe rather closer to where you define those get/post handlers. On the other hand the other approach would make routes aware of authentication which we don't want.

So... looks good to me! :)

**Comment by tomaszdurka** [comment]
Maybe you can bump version here too.

**Comment by vogdb** [comment]
@tomaszdurka please verify the versioning.

**Comment by tomaszdurka** [comment]
Looks good.

**Pull request merged**